infra/gravity/node_commands.go

```diff
@@ -155,14 +155,79 @@ type ClusterStatus struct {
 	Cluster string `json:"domain"`
 	// State is the cluster state
 	State string `json:"state"`
-	// SystemStatus is the cluster status, see https://github.com/gravitational/satellite/blob/7.1.0/agent/proto/agentpb/agent.proto#L50-L54
-	SystemStatus int `json:"system_status"`
+	// SystemStatus is the cluster status, custom unmarshalling logic needed to accommodate Gravity differences
+	SystemStatus int
 	// Token is secure token which prevents rogue nodes from joining the cluster during installation
 	Token Token `json:"token"`
 	// Nodes describes the nodes in the cluster
 	Nodes []NodeStatus `json:"nodes"`
 }
 
+// unmarshalSystemStatus is a helper type to parse string or integer system statuses.
+//
+// For the original definitions and mappings see:
+//   https://github.com/gravitational/satellite/blob/5.0.2/agent/proto/agentpb/agent.proto#L36-L40
+//   https://github.com/gravitational/satellite/blob/5.0.2/agent/proto/agentpb/agent.pb.go#L48-L63
+func unmarshalSystemStatus(val json.RawMessage, status *int) error {
+	err := json.Unmarshal(val, &status)
+	if err != nil { // parsing as int failed, try parsing as a string and converting
```

Comment:
In cases like this I would invert the condition to keep most of the code left-indented, it's a bit cleaner.

```go
err := json.Unmarshal(val, &status)
if err == nil {
  return nil
}
// rest of the code
```

Reply:
Addressed in a0af55e.